modules/developers-guide/pages/webpack-config.adoc

```diff
@@ -14,60 +14,104 @@ This section takes a closer look at the default front-end template, front-end co
+  // Interact with hello actor, calling the greet method
+  const greeting = await hello.greet(name);
+
+  document.getElementById("greeting").innerText = greeting;
+});
 ----
 
-After the agent and the actor are created for the project, the `+index.js+` file provides a placeholder for the JavaScript you use to interact with the document object model (DOM) and HTML for your application.
+In many projects, you will be able to ignore the code under `+declarations+` for the most part, and make your changes in `hello_assets/src`. However, if your project as additional requirements, continue reading below.
 
 === Modifying the webpack configuration
 
 Because webpack is a popular and highly-configurable module bundler for JavaScript-based applications, new projects create a default `+webpack.config.js+` file that makes it easy to add the specific modules—such as `react` and `markdown`—that you want to use.
 
-If you review the code in the template `+webpack.config.js+` file, you see that it constructs aliases for the canisters defined in your project's `+dfx.json+` file. The aliases can then be used to reference the canisters when imported as modules by the front-end.
+If you review the code in the template `+webpack.config.js+` file, you see that it infers canister ID's from your `.dfx/local/canister_ids.json` for local development, and from `'./canister_ids.json'` for any other environments you configure. It decides which network to use based on a `DFX_NETWORK` proccess variable, or based on whether `NODE_ENV` is set to `"production"`.
 
 You can see these steps in the following code block:
 
 [source,js]
 ----
-// List of all aliases for canisters. This creates the module alias for
-// the `import ... from "dfx-generated/canisters/xyz"` where xyz is the name of a
-// canister.
-const aliases = Object.entries(dfxJson.canisters).reduce(
-  (acc, [name, _value]) => {
-    // Get the network name, or `local` by default.
-    const networkName = process.env["DFX_NETWORK"] || "local";
-    const outputRoot = path.join(
-      __dirname,
-      ".dfx",
-      networkName,
-      "canisters",
-      name
-    );
-
-    return {
-      ...acc,
-      ["dfx-generated/" + name]: path.join(outputRoot, name + ".js"),
-    };
-  },
-  {}
-);
+let localCanisters, prodCanisters, canisters;
+
+try {
+  localCanisters = require(path.resolve(".dfx", "local", "canister_ids.json"));
+} catch (error) {
+  console.log("No local canister_ids.json found. Continuing production");
+}
+
+function initCanisterIds() {
+  try {
+    prodCanisters = require(path.resolve("canister_ids.json"));
+  } catch (error) {
+    console.log("No production canister_ids.json found. Continuing with local");
+  }
+
+  const network =
+    process.env.DFX_NETWORK ||
+    (process.env.NODE_ENV === "production" ? "ic" : "local");
+
+  canisters = network === "local" ? localCanisters : prodCanisters;
+
+  for (const canister in canisters) {
```

Comment:
Wont this error out if network === "local" and localCanisters is undefined?

Reply:
It will, but they would be encountering errors anyway if they were trying to call a local canister without having ever deployed their canister locally. "localCanisters is not defined" is a pretty useful error message to encounter